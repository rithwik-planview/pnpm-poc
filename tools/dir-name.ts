import { fileURLToPath } from 'url';
import { dirname } from 'path';

export const getDirname = (metaUrl: string) => dirname(fileURLToPath(metaUrl));
