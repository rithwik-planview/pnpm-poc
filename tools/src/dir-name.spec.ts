import { getDirname } from './dir-name';

describe('getDirName', () => {
    it('should return the correct directory name for each remote name', () => {
        expect(getDirname(import.meta.url)).toMatchSnapshot();
    });
});
