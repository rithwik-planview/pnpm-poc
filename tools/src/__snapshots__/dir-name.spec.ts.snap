// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDirName should return the correct directory name for each remote name 1`] = `"/Users/ranand/Code/module_federation/pnpm-poc/tools/src"`;
